concourse/scripts/kipper.sh

```diff
@@ -0,0 +1,361 @@
+  while [[ ${target_db_instances_count} -lt ${initial_db_instances_count} ]]; do
+    local target_db_instances
+    target_db_instances=$(get_db_instances "${target_db_state}" "${initial_db_instances_ids}")
+    local target_db_instances_ids
+    target_db_instances_ids=$(get_db_instances_ids "${target_db_instances}")
+    local target_db_instances_count
+    target_db_instances_count=$(get_length "${target_db_instances_ids}")
+    if [[ ${previous_db_instances_count} -ne ${target_db_instances_count} ]] || [[ "${previous_db_instances_count}" = '' ]]; then
+      echo "db instances ${target_db_state}: ${target_db_instances_count}/${initial_db_instances_count}"
+      previous_db_instances_count=${target_db_instances_count}
+      sleep 3
+    fi
+  done
+}
+
+print_error() {
+  local string
+  string="${1:-}"
+  local level
+  level="${2:-FATAL}"
+  echo -e "${level}: ${string}" >&2
+  if [[ "${level}" = 'FATAL' ]]; then
+    exit 1
+  fi
+}
+
+check_dependancies() {
+  echo 'checking dependancies...'
+  local bash_ver
+  bash_ver=$(echo "${BASH_VERSION}" | cut -d'.' -f1)
+  if [[ "${bash_ver}" -lt 4 ]]; then
+    print_error 'requires bash version >= 4'
+  fi
+  if ! command -v bosh >/dev/null 2>&1; then
+    print_error 'missing bosh cli'
+  fi
+  if ! command -v jq >/dev/null 2>&1; then
+    print_error 'missing jq'
+  fi
+  if ! command -v aws >/dev/null 2>&1; then
+    print_error 'missing aws cli'
+  fi
+}
+
+validate_initial_state() {
+  echo 'validating initial state...'
+  if ! bosh vms >/dev/null 2>&1; then
+    print_error 'not logged in to bosh'
+  fi
+  # make sure we are logged into the correct bosh env
+  if [[ "${BOSH_ENVIRONMENT:?}" != "bosh.${env_arg}.dev.cloudpipeline.digital" ]]; then
+    print_error "wrong bosh env targeted (expecting: ${env_arg})"
+  fi
+  if ! aws s3 ls >/dev/null 2>&1; then
+    print_error 'not logged in to aws'
+  fi
+  local ec2_instances
+  ec2_instances=$(get_ec2_instances)
+  local ec2_instances_ids
+  ec2_instances_ids=$(get_ec2_instances_ids "${ec2_instances}")
+  local ec2_instances_count
+  ec2_instances_count=$(get_length "${ec2_instances_ids}")
+  if [[ "${ec2_instances_count}" = '0' || "${ec2_instances_count}" = '' ]]; then
+    print_error "env does not exist: ${env_arg}"
+  fi
+  # check that ec2 instances are not already mid state change
+  local ec2_instances_transitioning
+  ec2_instances_transitioning=$(get_ec2_instances 'stopping,starting,pending,shutting-down')
+  local ec2_instances_transitioning_ids
+  ec2_instances_transitioning_ids=$(get_ec2_instances_ids "${ec2_instances_transitioning}")
+  local ec2_instances_transitioning_count
+  ec2_instances_transitioning_count=$(get_length "${ec2_instances_transitioning_ids}")
+  if [[ "${ec2_instances_transitioning_count}" != '0' || "${ec2_instances_transitioning_count}" = '' ]]; then
+    print_error "env is already in transition: ${env_arg}"
+  fi
+}
+
+parse_args() {
+  env_arg="${1:-}"
+  local regex="^dev([0-9]{2})$"
```

Comment:
We don't necessarily name our dev envs `devXX`, it would be good to still be able to spin up ones with other names and still have this script work.

Reply:
fixed